src/global-firestore-data.ts

```diff
@@ -198,6 +202,113 @@ export const deleteCourseFromSelectableRequirements = (courseUniqueID: number):
   );
 };
 
+export const setOnboardingData = (name: FirestoreUserName, onboarding: AppOnboardingData): void => {
+  usernameCollection
+    .doc(store.state.currentFirebaseUser.email)
+    .set({
+      firstName: name.firstName,
+      middleName: name.middleName || '',
+      lastName: name.lastName,
+    })
+    .then(() => store.commit('setUserName', name));
+  onboardingDataCollection
+    .doc(store.state.currentFirebaseUser.email)
+    .set({
+      gradYear: onboarding.gradYear,
+      entranceYear: onboarding.entranceYear,
+      colleges: [{ acronym: onboarding.college }],
+      majors: onboarding.major.map(acronym => ({ acronym })),
+      minors: onboarding.minor.map(acronym => ({ acronym })),
+      exam: onboarding.exam,
+      class: onboarding.transferCourse,
+      tookSwim: onboarding.tookSwim,
+    })
+    .then(() => {
+      const oldCollege = store.state.onboardingData.college;
+      store.commit('setOnboardingData', onboarding);
+      const newCollege = store.state.onboardingData.college;
+      if (oldCollege !== newCollege) {
+        clearOverridenRequirementsAPIB();
+      }
```

Comment:
I'm not sure that the `optOut` and `optIn` fields are being cleared when the college is changed. I wonder if `oldCollege` and `newCollege` might have the same value even in cases where I'm changing from CALS to A&S. Doesn't the store listen to changes on Firestore? So maybe by setting the `onboardingDataCollection` first, `oldCollege` would actually be the value of the college that you're changing to. Could you move `const oldCollege = store.state.onboardingData.college;` to before you set the onboarding doc?

https://user-images.githubusercontent.com/54298311/117287936-ef3bb000-ae38-11eb-8afe-fe7eb15df456.mov

Reply:
Oh that's so funny, I actually changed it in [a later commit](https://github.com/cornell-dti/course-plan/pull/489/commits/d58445441f3684219b5544466c403acbd04d269f) but didn't realize it would break it. Now that I think about it, because of the Firestore listeners, none of the `store.commit(...)` calls need to happen except for in `editSemesters()`, right? I think right now they're probably redundant.